doc/documentation.tex

```diff
@@ -466,6 +417,105 @@ \subsubsection{Methods comparison}
+
+\begin{figure}[H]
+	\centering
+    \subfloat[\centering CPD]{{\includegraphics[width=5cm]{mr-cpd-26.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering ICP]{{\includegraphics[width=5cm]{mr-icp-26.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering NICP]{{\includegraphics[width=5cm]{mr-nicp-26.png} }}
+    \caption{Tests with \textit{Config26.json}}
+\end{figure}
+
+CPD found exact solution. As mentioned above ICP has trouble with incomplete clouds and it remains unchanged with this configuration. NICP was close to finding exact solution; however, it lacks of $90\degree$ rotation.
+
+Finally we made test that only focus on situation where one cloud contains noise. We increased the \textit{noise-intensity} by small values and compared the results. The best method occurred to be NICP, CPD was the second and ICP was the third one. For tests, the configurations \textit{Config28.json} to \textit{Config38} were used. Results are presented below.
+
+\begin{figure}[H]
+	\centering
+    \subfloat[\centering CPD]{{\includegraphics[width=5cm]{mr-cpd-29.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering ICP]{{\includegraphics[width=5cm]{mr-icp-29.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering NICP]{{\includegraphics[width=5cm]{mr-nicp-29.png} }}
+    \caption{Tests with \textit{Config29.json}}
+\end{figure}
+
+\begin{figure}[H]
+	\centering
+    \subfloat[\centering CPD]{{\includegraphics[width=5cm]{mr-cpd-30.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering ICP]{{\includegraphics[width=5cm]{mr-icp-30.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering NICP]{{\includegraphics[width=5cm]{mr-nicp-30.png} }}
+    \caption{Tests with \textit{Config30.json}}
+\end{figure}
+
+\begin{figure}[H]
+	\centering
+    \subfloat[\centering CPD]{{\includegraphics[width=5cm]{mr-cpd-34.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering ICP]{{\includegraphics[width=5cm]{mr-icp-34.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering NICP]{{\includegraphics[width=5cm]{mr-nicp-34.png} }}
+    \caption{Tests with \textit{Config34.json}}
+\end{figure}
+
+\begin{figure}[H]
+	\centering
+    \subfloat[\centering CPD]{{\includegraphics[width=5cm]{mr-cpd-38.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering ICP]{{\includegraphics[width=5cm]{mr-icp-38.png} }}
+    \hspace{0.5em}
+    \subfloat[\centering NICP]{{\includegraphics[width=5cm]{mr-nicp-38.png} }}
+    \caption{Tests with \textit{Config38.json}}
+\end{figure}
+
+NICP managed to find exact solution in all test cases. Results obtained by CPD were slightly worse and sometimes correction of \textit{cpd-weight} parameter was needed. ICP results started to differ even when \textit{noise-intensity} was still very small, thus its usage when conditions are not perfect is very limited.
+
+To sum up the best algorithms for clouds with noises and outliers are NICP and CPD. It is worth mentioning that NICP is significantly faster than CPD. 
```

Comment:
Hmm, the last sentence somehow doesn't fit. Maybe something like: "In this situation, it is also worth mentioning that NICP was significantly faster then CPD in all the tests covered in this section". Something like this, not sure how to do it exactly but this sentence looks just like I've added it here to flex with NICP being the fastest 😎

Reply:
Done